### PR TITLE
[fix] fix venv_dev and multiprocess lib usage

### PR DESCRIPTION
**Pull request by bruntib**
Many functions and members of `multiprocess` lib are exposed dynamically. For this reason pylint is not able to tell what members this lib has (e.g. `multiprocess.Value()`. So these pylint reports have been suppressed.

Another purpose of this commit is to make "superset" relationship for venv_dev: if a CodeChecker developer installs venv_dev, then the content of venv should always be installed. venv_dev just extends them. `multiprocess` was a counterexample for this relationship: it was installed only by `venv` and not `venv_dev`. Such a mistake in the organization of requirements.txt files results missing libraries for developers.

### CODE REVIEW

File: codechecker_common/skiplist_handler.py [hunk 1]
```diff
@@ -103,12 +103,6 @@ def should_skip(self, source):
                 return sign == '-'
         return False
 
-    def __call__(self, source_file_path: str) -> bool:
```
Comment (vodorok) [inline codechecker_common/skiplist_handler.py:106]: Please do this in a separate pr.

File: codechecker_common/skiplist_handler.py [hunk 2]
```diff
@@ -117,11 +111,3 @@ def should_skip(self, file_path: str):
         handler.
         """
         return any(handler.should_skip(file_path) for handler in self)
-
```
Comment (vodorok) [inline codechecker_common/skiplist_handler.py:120]: Please do this in a separate pr.

File: tools/report-converter/codechecker_report_converter/report/__init__.py [hunk 3]
```diff
@@ -14,15 +14,16 @@
...
 
 from .. import util
 
 
 LOG = logging.getLogger('report-converter')
 
 
-SkipListHandlers = Callable[[str], bool]
+class SkipListHandlers(Protocol):
+    should_skip: Callable[[str], bool]
```
Comment (vodorok) [inline tools/report-converter/codechecker_report_converter/report/__init__.py:26]: Please do this in a separate pr.

File: tools/report-converter/codechecker_report_converter/report/reports.py [hunk 4]
```diff
@@ -17,8 +17,8 @@
 
 
 class GenericSuppressHandler:
-    get_suppressed: Callable[[Any, Report], bool]
-    store_suppress_bug_id: Callable[[Any, str, str, str, str], bool]
+    get_suppressed: Callable[[Report], bool]
```
Comment (vodorok) [inline tools/report-converter/codechecker_report_converter/report/reports.py:20]: Please do this in a separate pr.

File: tools/report-converter/codechecker_report_converter/report/__init__.py [hunk 5]
```diff
@@ -473,7 +474,7 @@ def skip(self, skip_handlers: Optional[SkipListHandlers]) -> bool:
         if not skip_handlers:
             return False
 
-        return skip_handlers(self.file.original_path)
+        return skip_handlers.should_skip(self.file.original_path)
```
Comment (vodorok) [inline tools/report-converter/codechecker_report_converter/report/__init__.py:477]: Please do this in a separate pr.

**Review by vodorok — CHANGES_REQUESTED** [review]
Please mention the pylint fixes in the commit title.
I think the skiplist refactoring should be done in a separate pr.

File: tools/report-converter/codechecker_report_converter/report/reports.py (hunk 4 above)
Comment (bruntib) [reply tools/report-converter/codechecker_report_converter/report/reports.py:20]: This change is related to mypy upgrade, so I would leave it here.
